api/command/mixins/team.py

```diff
@@ -0,0 +1,503 @@
+                          f"to {gh_team_name}")
+            try:
+                channel_member_ids = \
+                    self._slack_client.get_channel_users(channel)
+                logging.debug(f"Member IDs of members found in #{channel}: "
+                              f"{channel_member_ids}")
+            except SlackAPIError as e:
+                msg = f"Channel member query on channel #{channel} failed: " \
+                    f"{e.error}"
+                logging.error(msg)
+                raise SlackAPIError(msg)
+            for member_id in channel_member_ids:
+                try:
+                    member = self._db_facade.retrieve(User, member_id)
+                    self._gh_interface.add_team_member(member.github_username,
+                                                       gh_team_id)
+                    team.add_member(member.github_id)
+                    logging.debug(f"Member with ID {member.slack_id} added "
+                                  f"to {gh_team_name}")
+                except LookupError:
+                    logging.warning(f"Member with ID {member_id} not found, "
+                                    "ignoring...")
+                    pass
+        else:
+            self._gh_interface.add_team_member(command_user.github_username,
+                                               gh_team_id)
+            team.add_member(command_user.github_id)
+            logging.debug(f"Calling user with ID {command_user.slack_id} "
+                          f"added to {gh_team_name}")
+
+        if lead_id is not None:
+            lead = self._db_facade.retrieve(User, lead_id)
+
+            if check_permissions(lead, None):
+                lead_in_team = self._gh_interface.has_team_member(
+                    lead.github_username,
+                    gh_team_id)
+                if not lead_in_team:
+                    self._gh_interface.add_team_member(lead.github_username,
+                                                       gh_team_id)
+
+                team.add_member(lead.github_id)
+                team.add_team_lead(lead.github_id)
+                logging.debug(f"User with ID {lead_id} set as tech lead of "
+                              f"{gh_team_name}")
+            else:
+                msg = f"User specified with lead ID {lead_id} has S" \
+                    f" permission level {str(lead.permissions_level)}, " \
+                    "insufficient to lead a team!"
+                logging.error(msg)
+                raise PermissionError(msg)
+        else:
+            team.add_team_lead(command_user.github_id)
+            logging.debug(f"Calling user with ID {command_user.github_id} set"
+                          f" as tech lead of {gh_team_name}")
+
+        created = self._db_facade.store(team)
+        if created:
+            logging.info(f"Team succesfully created: {team.__str__()}")
+        else:
+            logging.error("Team creation unsuccessful")

```

Comment:
I guess, a bit of a misnomer, because this only checks if the team is stored into DDB, but doesn't check if it has been created on the Github side. Maybe rename to `Team successfully stored`.

Reply:
Actually, the database logs if the store is successful or not. No need to log it here.

Reply:
Removed